File: Backend/MedicineDelivery/MedicineDelivery.Domain/Enums/BusinessType.cs
namespace MedicineDelivery.Domain.Enums
{
    /// <summary>
    /// Legal business type of a chemist (medical store), collected as part of KYC.
    /// Numeric values are fixed to match the reference KYC form's dropdown codes.
    /// </summary>
    public enum BusinessType
    {
        Proprietorship = 1,
        Partnership = 3,
        PrivateLimited = 4,
        PublicLimited = 5,
        LLP = 6,
        NGO = 7,
        Trust = 9,
        Society = 10,
        Individual = 11
    }
}

File: Flutter_UI/lib/shared/models/business_type.dart
/// Legal business type of a chemist (medical store), collected as part of KYC.
/// Values mirror the backend's `MedicineDelivery.Domain.Enums.BusinessType` enum.
enum BusinessType {
  proprietorship(1, 'Proprietorship'),
  partnership(3, 'Partnership'),
  privateLimited(4, 'Private Limited'),
  publicLimited(5, 'Public Limited'),
  llp(6, 'LLP'),
  ngo(7, 'NGO'),
  trust(9, 'Trust'),
  society(10, 'Society'),
  individual(11, 'Individual');

  final int value;
  final String label;

  const BusinessType(this.value, this.label);

  static BusinessType fromValue(int? value) {
    return BusinessType.values.firstWhere(
      (t) => t.value == value,
      orElse: () => BusinessType.privateLimited,
    );
  }
}
